### DISCUSSION TITLE
[Run issue]: Gvalue for a simulation including a magnetic field

### DISCUSSION THREAD

**Discussion by Audrey622** [topas-nbio/TOPAS-nBio#25] [Run-Problem] [answered]
### What was the issue?

Hi!

I'm trying to obtain Gvalues for the simulation of a beam of electrons passing through a geometry made of G4_WATER in the presence of a magnetic field. However, the following error is being returned:

(...)
G4WT1 > Begin processing for Run: 0, History: 0
G4WT1 >
-------- EEEE ------- G4Exception-START -------- EEEE -------
*** G4Exception : NoExternalFieldSupport
      issued by : G4ITTransportation::AlongStepGetPhysicalInteractionLength
ITTransportation does not support external fields. If you are dealing with a tradiational MC simulation, please use G4Transportation.
*** Fatal Exception *** core dump ***
G4WT1 > G4Track (0x752fbd4e0300) - track ID = -19, parent ID = -6
(...)

Is there a way to obtain the Gvalue for this simulation?

### TOPAS-nBio category

Scoring

### Version

3.0

### What Operative System (OS) are you running on?

Linux (Debian/Ubuntu)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by JoseRamosMendez** [comment, marked as answer]
In the current implementation of the chemistry, species don't interact with magnetic fields. Perhaps you could save the electron track's phase space simulated in the presence of a magnetic field and then use it for the G-value calculation.
Good luck!

**Reply by Audrey622** [reply]
Thank you a lot for your answer! It was all I needed to know!
